src/test/regress/sql/multi_extension.sql

```diff
@@ -400,13 +400,21 @@ SELECT * FROM multi_extension.print_extension_changes();
 ALTER EXTENSION citus UPDATE TO '10.2-4';
 SELECT * FROM multi_extension.print_extension_changes();
 
--- Snapshot of state at 10.2-5
-ALTER EXTENSION citus UPDATE TO '10.2-5';
+-- There was a bug when downgrading to 10.2-2 from 10.2-4
+-- Test that we do not have any issues with this particular downgrade
+ALTER EXTENSION citus UPDATE TO '10.2-2';
+ALTER EXTENSION citus UPDATE TO '10.2-4';
```

Comment:
Above we test a downgrade from `10.2-4` to `10.2-3` and a separate one for `10.2-3` to `10.2-2`. How is a direct downgrade from `10.2-4` to `10.2-2` causing an issue that we didn't run into above?

and actually there is a test:
```
-- Make sure that we don't delete pg_depend entries added in
-- columnar--10.2-3--10.2-4.sql when downgrading to 10.2-3.
SELECT COUNT(*)=10
FROM pg_depend
WHERE classid = 'pg_am'::regclass::oid AND
      objid = (select oid from pg_am where amname = 'columnar') AND
      objsubid = 0 AND
      refclassid = 'pg_class'::regclass::oid AND
      refobjsubid = 0 AND
      deptype = 'n';
```

Which seems to be specifically testing the invariant you are changing here.

To say the least, I am confused.